Adding info for CORS headers (#47)
AdobeDocs/graphql-mesh-gateway: src/pages/gateway/headers.md

# Headers

To specify request headers for your mesh, you can add them inside the `JSON` file that describes your mesh, or you can add them when querying. Currently, you can add [request headers](#request-headers) and [response headers](#response-headers).

## Request headers

Request headers provide more information about the request context. Currently, you can add request headers to your [mesh config](#add-request-headers-in-your-mesh-file) or add them [at runtime](#add-or-update-request-headers-at-runtime).

<InlineAlert variant="info" slots="text"/>

> All header names must be unique. Conflicting header names will result in an error.

Reviewer: Yes, that's right, only response headers need to be unique.

-  **Key**: `GGW-SH-differentSource-trackingId`
   -  **Value**: `different-trackingId`

## Response headers

Response headers provide detailed context of the response. Currently, you can only add response headers from the mesh config.

<InlineAlert variant="info" slots="text"/>

> All response header names must be unique. Conflicting header names will result in an error on mesh creation and update.

Reviewer 2: ```suggestion
All response header names must be unique. Conflicting header names will result in an error when you try to create or update a mesh.
```

-  **Key**: `GGW-SH-differentSource-trackingId`
   -  **Value**: `different-trackingId`

## Response headers

Response headers provide detailed context of the response. Currently, you can only add response headers from the mesh config.

<InlineAlert variant="info" slots="text"/>

> All header names must be unique. Conflicting header names will result in an error.

Reviewer 3: ```suggestion
All header names must be unique. Conflicting header names will result in an error on mesh creation and update.
```

Also FYI, this won't be true until we implement [CEXT-30](https://jira.corp.adobe.com/browse/CEXT-30) (and only for response headers configured at the source level)

      }
    }
```

#### CORS headers

> Cross-origin resource sharing or CORS allows you to pass usually restricted resources to an outside domain. Refer to [MDN's documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) for more information on CORS headers.

Reviewer 2: ```suggestion
Cross-origin resource sharing (CORS) allows you to pass resources that are usually restricted to an outside domain. Refer to [MDN's documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) for more information on CORS headers.
```

To add CORS headers to your mesh, create a `CORS` object in the `responseConfig` object, using the following key value pairs:

-  `origin` (Required) - the domain of the resource you want to allow to make a CORS request

> -  `methods` (Required) - the HTTP verbs allowed in the CORS request

Reviewer 2: They're not all verbs, such as OPTIONS. I went to the MDN docs, and I'm inferring  that the list of possible values are in the left nav. Consider changing the description to the following. You might want to have some different request methods than what's in the example.

```suggestion
-  `methods` (Required) - the HTTP request methods allowed in the CORS request, such as GET, POST, and OPTIONS
```

      }
    }
```

> #### CORS headers

Reviewer 3: This section LGTM, but @of0x would know better than I.
